compressor/conf.py

```diff
@@ -131,3 +141,30 @@ def configure_precompilers(self, value):
                                        "must be a list or tuple. Check for "
                                        "missing commas.")
         return value
+
+    def configure(self):
+        data = self.configured_data
+        for kind in {'css', 'js'}:
+            setting_name = '%s_FILTERS' % kind.upper()
+            filters = data[setting_name]
+            if filters is not None:
+                # filters for this kind are set using <kind>_FILTERS
+                if kind in data['FILTERS']:
+                    raise ImproperlyConfigured(
+                        "The setting {kind_setting} "
+                        "conflicts with {main_setting}['{kind}']. "
+                        "Remove either setting and update the other to "
+                        "the correct list of filters for {kind} resources"
+                        ", we recommend you keep the latter."
+                        .format(
+                            kind_setting=self._meta.prefixed_name(setting_name),
+                            main_setting=self._meta.prefixed_name('FILTERS'),
+                            kind=kind))
+                data['FILTERS'][kind] = filters
+            elif kind in data['FILTERS']:
+                # filters for this kind are set using FILTERS[<kind>]
+                data[setting_name] = data['FILTERS'][kind]
+            else:
+                # filters are not defined
+                data['FILTERS'][kind] = data[setting_name] = default_filters[kind]
+        return data
```

Comment:
i would rather settle on one format and let this code here transfer the other format if necessary, instead of ensuring both formats are present. what do you think?

Reply:
I was going for something that supports both formats
during the transition, mainly since it makes the
code and testing a bit nicer IMO.

Reply:
i don't quite understand, if the code outside of conf.py accessed only one of the two options (which it should in my opinion), then of what use is it to ensure the other one is present as well?

as to backwards-compatibility, the idea is of course to be backwards-compatible. 

i feel like we are misunderstanding each other, here an attempt to clarify my idea: just like it does right now in your PR, conf.py should be able to read both formats (while ensuring that only one is actually present to avoid conflicts), but in contrast to your PR, it should "output" only one format. the code outside of conf.py would read only that format. in my understanding, it should be strictly less code than the current PR.

Reply:
I implemented the change as you suggested. The old form is no longer present after configuring.
Thanks for clarifying.